client/asset/eth/eth.go

```diff
@@ -3484,540 +3718,189 @@ func (eth *ETHWallet) checkForNewBlocks(ctx context.Context) {
+	return &asset.ConfirmRedemptionStatus{
+		Confs:  confs,
+		Req:    req,
+		CoinID: txHash[:],
 	}
-	return nil, fmt.Errorf("there is a cycle in the monitored transactions")
 }
 
-// recordReplacementTx updates a monitoredTx with a replacement transaction.
-// This change is also stored in the db.
-//
-// originalTx's mtx must be held when calling this function.
-func (w *assetWallet) recordReplacementTx(originalTx *monitoredTx, replacementHash common.Hash) error {
-	originalTx.replacementTx = &replacementHash
-	originalHash := originalTx.tx.Hash()
-	if err := w.txDB.storeMonitoredTx(originalHash, originalTx); err != nil {
-		return fmt.Errorf("error recording replacement tx: %v", err)
+// confirmRedemption checks the confirmation status of a redemption transaction.
+func (w *assetWallet) confirmRedemption(coinID dex.Bytes, redemption *asset.Redemption) (*asset.ConfirmRedemptionStatus, error) {
+	if len(coinID) != common.HashLength {
+		return nil, fmt.Errorf("expected coin ID to be a transaction hash, but it has a length of %d",
+			len(coinID))
 	}
+	var txHash common.Hash
+	copy(txHash[:], coinID)
 
-	w.monitoredTxsMtx.Lock()
-	defer w.monitoredTxsMtx.Unlock()
-
-	replacementTx, found := w.monitoredTxs[replacementHash]
-	if !found {
-		w.log.Errorf("could not find replacement monitored tx %s", replacementHash)
-	}
-	replacementTx.mtx.Lock()
-	defer replacementTx.mtx.Unlock()
-	replacementTx.replacedTx = &originalHash
-	if err := w.txDB.storeMonitoredTx(replacementHash, replacementTx); err != nil {
-		return fmt.Errorf("error recording replaced tx: %v", err)
+	contractVer, secretHash, err := dexeth.DecodeContractData(redemption.Spends.Contract)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode contract data: %w", err)
 	}
 
-	return nil
-}
-
-// txsToDelete retraces the doubly linked list to find the all the
-// ancestors of a monitoredTx.
-func (w *assetWallet) txsToDelete(tx *monitoredTx) []common.Hash {
-	txsToDelete := []common.Hash{tx.tx.Hash()}
+	tip := w.tipHeight()
 
-	maxLoops := 100 // avoid an infinite loop in case of a cycle
-	for i := 0; i < maxLoops; i++ {
-		if tx.replacedTx == nil {
-			return txsToDelete
+	// If we have local information, use that.
+	if found, s := w.localTxStatus(txHash); found {
+		if s.assumedLost || len(s.nonceReplacement) > 0 {
+			if s.nonceReplacement == s.feeReplacement {
+				// Tell core to update it's coin ID.
+				txHash = common.HexToHash(s.feeReplacement)
+			} else {
+				return nil, asset.ErrTxLost
+			}
 		}
-		txsToDelete = append(txsToDelete, *tx.replacedTx)
-		var found bool
-		tx, found = w.monitoredTxs[*tx.replacedTx]
-		if !found {
-			w.log.Errorf("failed to find replaced tx: %v", *tx.replacedTx)
-			return txsToDelete
+		// Should we require > 1 confirmation on rejected txs before reporting?
+		if s.receipt != nil && s.receipt.Status != types.ReceiptStatusSuccessful {
+			return nil, asset.ErrTxRejected
+		}
```

Comment:
Might as well wait for `FinalizeConfs`. We could add a `Rejected` field to `asset.WalletTransaction` and then that could also be displayed in the transaction history.